File: scripts/test_backend_injection.ts
import { createClient } from '@insforge/sdk';

const insforgeUrl = 'https://sxwv82iw.us-east.insforge.app';
const insforgeKey = 'ik_12002a3fd3274a14e562bcce4a015fee';

const insforge = createClient({ baseUrl: insforgeUrl, anonKey: insforgeKey });

async function main() {
    console.log("Iniciando sesión...");
    const { data: authData, error: authError } = await insforge.auth.signInWithPassword({
        email: "elevateenterprisebrands@gmail.com",
        password: "Ww172839456*-@"
    });

    if (authError || !authData?.user) {
        console.error("Error en login:", authError);
        return;
    }

    const userId = authData.user.id;
    console.log("Sesión iniciada exitosamente! User ID:", userId);

    for (let i = 1; i <= 3; i++) {
        console.log(`Creando cliente de prueba ${i}...`);
        
        const clientData = {
            lender_id: userId,
            name: `Cliente de Prueba ${i}`,
            cedula: `402-${Math.floor(1000000 + Math.random() * 9000000)}-${Math.floor(1 + Math.random() * 9)}`,
            email: `cliente${i}@test.com`,
            phone: `809-555-${Math.floor(1000 + Math.random() * 9000)}`,
            address: `Calle Falsa ${i}, Santo Domingo`,
            occupation: "Empleado",
            income: Math.floor(15000 + Math.random() * 35000),
            status: "Activo",
            joineddate: new Date().toISOString().split('T')[0]
        };

        const { data: insertedClient, error: clientError } = await insforge.database
            .from('clients')
            .insert(clientData)
            .select()
            .single();

        if (clientError || !insertedClient) {
            console.error("Error creando cliente:", clientError);
            continue;
        }

        const clientId = (insertedClient as any).id;
        
        console.log(`Creando préstamo para Cliente de Prueba ${i}...`);
        
        const amount = Math.floor(5000 + Math.random() * 15000);
        const durationWeeks = [12, 24, 36][Math.floor(Math.random() * 3)];
        const interestRate = [5, 10, 15][Math.floor(Math.random() * 3)];
        const totalToPay = amount + (amount * (interestRate / 100));

        const loanData = {
            lender_id: userId,
            clientid: clientId,
            clientname: `Cliente de Prueba ${i}`,
            amount: amount,
            interestrate: interestRate,
            durationweeks: durationWeeks,
            frequency: "Mensual",
            startdate: new Date().toISOString().split('T')[0],
            status: "Activo",
            installmentamount: totalToPay / durationWeeks,
            remainingbalance: totalToPay,
            totaltopay: totalToPay,
            loantype: "Amortizado",
            collateraltype: "Sin Garantía"
        };

        const { error: loanError } = await insforge.database
            .from('loans')
            .insert(loanData);

        if (loanError) {
            console.error("Error creando préstamo:", loanError);
        } else {
            console.log("Préstamo creado con éxito!");
        }
    }
    
    console.log("¡Inyección de datos finalizada correctamente!");
}

main();

File: scripts/test_backend_injection.py
import requests
import json
import random
from datetime import datetime, timedelta

# Configuration
BASE_URL = "https://sxwv82iw.us-east.insforge.app"
API_KEY = "ik_12002a3fd3274a14e562bcce4a015fee"
EMAIL = "elevateenterprisebrands@gmail.com"
PASSWORD = "Ww172839456*-@"

# Headers
HEADERS = {
    "apikey": API_KEY,
    "Content-Type": "application/json"
}

def login():
    url = f"{BASE_URL}/auth/v1/token?grant_type=password"
    payload = {
        "email": EMAIL,
        "password": PASSWORD
    }
    print("Iniciando sesión...")
    response = requests.post(url, headers=HEADERS, json=payload)
    if response.status_code == 200:
        data = response.json()
        print("Sesión iniciada exitosamente!")
        return data["access_token"], data["user"]["id"]
    else:
        print("Error en login:", response.text)
        return None, None

def create_client(token, user_id, index):
    url = f"{BASE_URL}/rest/v1/clients"
    headers = {**HEADERS, "Authorization": f"Bearer {token}"}
    
    client_data = {
        "lender_id": user_id,
        "name": f"Cliente de Prueba {index}",
        "cedula": f"402-{random.randint(1000000, 9999999)}-{random.randint(1,9)}",
        "email": f"cliente{index}@test.com",
        "phone": f"809-555-{random.randint(1000, 9999)}",
        "address": f"Calle Falsa {index}, Santo Domingo",
        "occupation": "Empleado",
        "income": random.randint(15000, 50000),
        "status": "Activo"
    }
    
    # Supabase uses Prefer: return=representation to get inserted row back
    headers["Prefer"] = "return=representation"
    
    print(f"Creando cliente {index}...")
    response = requests.post(url, headers=headers, json=client_data)
    if response.status_code in [200, 201]:
        return response.json()[0]["id"]
    else:
        print("Error creando cliente:", response.text)
        return None

def create_loan(token, user_id, client_id, client_name):
    url = f"{BASE_URL}/rest/v1/loans"
    headers = {**HEADERS, "Authorization": f"Bearer {token}", "Prefer": "return=representation"}
    
    amount = random.randint(5000, 20000)
    duration_weeks = random.choice([12, 24, 36])
    interest_rate = random.choice([5, 10, 15])
    total_to_pay = amount + (amount * (interest_rate / 100))
    
    loan_data = {
        "lender_id": user_id,
        "clientId": client_id,
        "clientName": client_name,
        "amount": amount,
        "interestRate": interest_rate,
        "durationWeeks": duration_weeks,
        "frequency": "Mensual",
        "startDate": datetime.now().strftime("%Y-%m-%d"),
        "status": "Activo",
        "installmentAmount": total_to_pay / duration_weeks,
        "remainingBalance": total_to_pay,
        "totalToPay": total_to_pay,
        "loanType": "Amortizado",
        "collateralType": "Sin Garantía"
    }
    
    print(f"Creando préstamo para {client_name}...")
    response = requests.post(url, headers=headers, json=loan_data)
    if response.status_code in [200, 201]:
        return response.json()[0]["id"]
    else:
        print("Error creando préstamo:", response.text)
        return None

def main():
    token, user_id = login()
    if not token:
        return
    
    # Crear 3 clientes de prueba
    for i in range(1, 4):
        client_id = create_client(token, user_id, i)
        if client_id:
            # Crear un préstamo para cada cliente
            create_loan(token, user_id, client_id, f"Cliente de Prueba {i}")

if __name__ == "__main__":
    main()
